tests/integration/test_s3.py

```diff
@@ -1099,3 +1099,36 @@ def test_redirects_head_object(self):
             self.assertEqual(response.get('ContentLength'), len(key))
         except ClientError as e:
             self.fail("S3 Client failed to redirect Head Object: %s" % e)
+
+    def test_create_bucket_without_location_uses_us_east_1(self):
+        bucket_name = random_bucketname()
+        response = self.client.create_bucket(Bucket=bucket_name)
+        self.assert_status_code(response, 200)
+        self.addCleanup(recursive_delete, self.client, bucket_name)
+
+        waiter = self.client.get_waiter('bucket_exists')
+        waiter.wait(Bucket=bucket_name)
+
+        response = self.client.head_bucket(Bucket=bucket_name)
+        headers = response['ResponseMetadata']['HTTPHeaders']
+        region = headers.get('x-amz-bucket-region', None)
+        self.assertEqual(region, 'us-east-1')
+
+    def test_create_bucket_with_location_does_not_redirect(self):
```

Comment:
I'm not entirely sure what you're saying. We redirect if location constraint isn't set. We do not redirect if it is set. In this test I just happen to use `self.client_region` because it is readily available.